cryptol-saw-core/src/Verifier/SAW/Cryptol.hs

```diff
@@ -1648,11 +1681,19 @@ proveEq sc env t1 t2
         | map fst (C.canonicalFields tm1) == map fst (C.canonicalFields tm2) ->
           proveEq sc env (C.tTuple (map snd (C.canonicalFields tm1))) (C.tTuple (map snd (C.canonicalFields tm2)))
 
-      -- XXX: add a case for `enum`
-      -- 1. Match constructors by names, and prove fields as tuples
-      -- 2. We need some way to combine the proofs of equality of
-      -- the fields, into a proof for equality of the whole type
-      -- for sums
+      (C.tIsNominal -> Just (C.NominalType{C.ntDef=C.Enum _},_),
+       C.tIsNominal -> Just (C.NominalType{C.ntDef=C.Enum _},_)) ->
+        panic "proveEq" ["Enum types unsupported.", pretty t1, pretty t2]
+
+        -- XXX: add a case for `enum`
+        -- 1. Match constructors by names, and prove fields as tuples
+        -- 2. We need some way to combine the proofs of equality of
+        -- the fields, into a proof for equality of the whole type
+        -- for sums
+        --
+        -- XXX: Response to above: Not sure what purpose of `proveEq`
+        -- is, but wouldn't Enum types have name (not structural)
+        -- equality?
```

Comment:
Among other things, `proveEq` is used in the implementation of `coerceTerm`, which powers the `fallback` function mentioned above. If I understand correctly, one way to test this would be to try and import the following Cryptol code into SAW:

```
enum Letter = A | B

f l =
  case l of
    A => B
    B => A
```

This has a `case` expression where the inferred type of the expression is an enum. I'm curious to know if that panics when using this PR.

Reply:
it does not panic.
TODO for me
- teence exploring to find something that induces the panic
- create an issue to: either induce the panic or determine what circumstances could cause the panic

Reply:
I have having trouble myself with trying to come up with a test case, which demonstrates that I don't understand this code as well as I thought I did :)

In any case, perhaps we should wait until we actually encounter this `panic` in the wild before taking action here.